src/lib/image-upload-provider.ts

```diff
@@ -0,0 +1,53 @@
+import { ImageUploadProvider } from '@atlaskit/editor-common/dist/types/provider-factory'
+import { ElementType, Errors, EventType, InputType } from 'constants/strings'
+import { processFile } from 'lib/utils/process-file'
+
+interface CollabProvider {
+  config: {
+    noteId: string
+  }
+  serviceClient: {
+    postImage: (
+      name: string,
+      noteId: string,
+      file: ArrayBuffer
+    ) => Promise<{ data: { id: string } }>
+  }
+}
+
+export const imageUploadProvider = (
+  collabProvider: CollabProvider
+): Promise<ImageUploadProvider> =>
+  Promise.resolve<ImageUploadProvider>((_event, insertImageFn) => {
+    const inputElement = document.createElement(ElementType.Input)
+
+    inputElement.type = InputType.File
```

Comment:
You can add an `accept="image/*"` attribute on the input element. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#attr-accept

Reply:
Yes oversight from my part, we have to restrict file type

Reply:
Should handle rejection UI side though, I will create a ticket; updated